docs/docs/user/core-concepts/operation-mode.md

```diff
@@ -0,0 +1,136 @@
+# Operation Modes in KPOps
+
+KPOps supports three operation modes—`managed`, `manifest`, and `argo`. These modes determine how resources are managed and allow users to tailor their deployment strategy.
+
+- **Managed Mode** (default): KPOps uses helm under the hood to manage the installation of applications in the defined pipeline.yaml.
+- **Manifest Mode**: Focuses on generating Kubernetes manifests.
+- **Argo Mode**: Extends the functionality to include ArgoCD-specific hooks for certain operations, facilitating GitOps workflows with automated cleanup and reset tasks.
+
+---
+
+## Configuring Operation Modes
+
+### `manifest` Mode
+
+Enable `manifest` mode by setting the following in your configuration:
+
+```yaml
+operation_mode: manifest
+```
+
+### `argo` Mode
+
+Enable `argo` mode by setting the following in your configuration:
+
+```yaml
+operation_mode: argo
+```
+
+<!-- dprint-ignore-start -->
+
+!!! info Set operation-mode over CLI
+     It is possible to pass the `--operation-mode <OPERATION>` option in the CLI to override the operation mode of the `config.yaml`. You can refer to the [CLI commands documentation](https://bakdata.github.io/kpops/9.0/user/references/cli-commands/#kpops-deploy) for more details.
+
+<!-- dprint-ignore-end -->
+
+---
+
+## Generated Resources by Mode and Operation
+
+### `deploy`
+
+#### **Manifest Mode**
+
+- **Streams bootstrap Applications**:
+  - Depending on your pipeline configuration, Kubernetes `Job`, `Deployment`, `ConfigMap`, and `Service` resources.
+  - Please refer to [Streams-bootstrap Helm Charts](https://github.com/bakdata/streams-bootstrap/tree/master/charts).
+- **Topics**:
+  - Strimzi `KafkaTopic` CRDs.
+
+#### **Argo Mode**
+
+- **Streams bootstrap Applications**:
```

Comment:
please stick to one consistent naming, I believe the official name is _streams-bootstrap_